src/utils/api.js

```diff
@@ -4,4 +4,59 @@ const api = axios.create({
     baseURL: "https://nc-news-p6kr.onrender.com/api/"
 })
 
-export default api
\ No newline at end of file
+const getAllArticles = (url, sort, order) => {
+
+    const getArticles = async () => {
+        try {
+            const response = await api.get(url)
+            let articles = response.data.articles
+            if(sort === 'created_at'){
+                if(order === 'asc'){
+                    articles.sort((a, b) => Date.parse(a[sort]) - Date.parse(b[sort]))
+                } else {
+                    articles.sort((a, b) => Date.parse(b[sort]) - Date.parse(a[sort]))
+                }
+            } else {
+                if(order === 'asc'){
+                    articles.sort((a, b) => a[sort] - b[sort])
+                } else {
+                    articles.sort((a, b) => b[sort] - a[sort])
+                }
+            }
+            return(response.data.articles)
+        }catch(err) {
```

Comment:
You are currently fetching all the articles and then sorting them client side. This should be server side. We built the back end so it could accept sort by and order by queries.

This won't make an impact on a programme this small, however it will make a significant impact the moment you start dealing with a lot of data.

This needs refactoring asap